Server/src/main/java/jettyServer/resources/indexingResource/IndexingResource.java

```diff
@@ -0,0 +1,63 @@
+package jettyServer.resources.indexingResource;
+
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import javax.inject.Inject;
+import javax.ws.rs.core.Response;
+
+import jettyServer.configuration.ServerConfiguration;
+import kafka.ObjectSerializer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+@Path("index")
+public class IndexingResource {
+    private final ServerConfiguration serverConfiguration;
+    private final  Producer producer;
+
+
+    @Inject
+    public IndexingResource(ServerConfiguration serverConfiguration, Producer producer){
+        this.serverConfiguration = serverConfiguration;
+        this.producer=producer;
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response post(IndexingRequestObj body, @HeaderParam("user-agent") String userAgent) {
+        Map source = createSource(body.getMessage(),userAgent);
+        String topic = "bootcamp";
+
+        ProducerRecord<String, Map> record = new ProducerRecord<>(topic, source);
+        try{
+            producer.send(record).get();
+            return Response.ok().entity("OK").build();
+        }
+        catch (Exception e){
+            return Response.status(Response.Status.BAD_REQUEST).entity(e.getCause()).build();
```

Comment:
I believe you don't want to respond the client with `e.getCause()` as it can be internal errors in kafka. Let's do `Logger.error` with the exception, and respond with a friendly message to the client